libsolidity/ast/Types.cpp

```diff
@@ -2138,17 +2139,11 @@ TypePointer StructType::interfaceType(bool _inLibrary) const
 
 bool StructType::canBeUsedExternally(bool _inLibrary) const
 {
-	if (_inLibrary && location() == DataLocation::Storage)
-		return true;
-	else if (recursive())
+	if (recursive())
 		return false;
 	else
 	{
-		// Check that all members have interface types.
-		// We pass "false" to canBeUsedExternally (_inLibrary), because this struct will be
-		// passed by value and thus the encoding does not differ, but it will disallow
-		// mappings.
-		// Also return false if at least one struct member does not have a type.
+		// Return false if at least one struct member does not have a type.
```

Comment:
This is only half of what's happening - I think you should keep the first line of the original comment.